### DISCUSSION TITLE
Explicitly set a specific reader class for markdown files

### DISCUSSION THREAD

**Discussion by frederik-elwert** [getpelican/pelican#3428] [Q&A] [answered]
My initial problem is that I want to use both the [jinja2content](https://github.com/pelican-plugins/jinja2content) plugin and the [yaml-metadata](https://github.com/pelican-plugins/yaml-metadata) plugin. Since they both register themselves as readers for Markdown files, unfortunately their effect is not combined, but rather only one of them is active at a time (in my case it was the YAML metadata plugin that “won”).

My attempt at a solution was creating a simple new reader that subclasses both:

```python
from pelican.plugins.jinja2content import JinjaContentMixin
from pelican.plugins.yaml_metadata import YAMLMetadataReader

class JinjaYAMLMarkdownReader(JinjaContentMixin, YAMLMetadataReader):
    pass
```

I tried activating that in the settings:

```python
READERS = {"md": JinjaYAMLMarkdownReader}
```

Unfortunately, this does not have an effect. What did help, though, is registering a new format, "mdj", and rename my source files accordingly. My interpretation is that the READERS setting does not actually set the readers, but just adds the specified class to the internal list, but another registered reader wins.

Is my understanding correct? Is there a way to explicitly set a reader class for a format? Or would there be a smarter way to solve this?

**Comment by dltj** [comment, marked as answer]
Frederick — I had the [same exact problem](https://github.com/getpelican/pelican/discussions/3373)...take a look at [my solution](https://github.com/dltj/zygote-reader) for ideas on how to address it.

**Reply by frederik-elwert** [reply]
Thanks! So the trick really was to name the plugin accordingly? It seems a bit overkill to create a full plugin if a simple class definition would do the trick, but if that’s the only way to override the loading order, then so be it …
